include/manif/impl/rn/Rn_base.h

```diff
@@ -0,0 +1,226 @@
+   * @param  v A 2-vector.
+   * @param[out] -optional- J_vout_m The Jacobian of the new object wrt this.
+   * @param[out] -optional- J_vout_v The Jacobian of the new object wrt input object.
+   * @return The rotated 2-vector.
+   * @note See Eqs. (129, 130).
+   */
+  template <typename _EigenDerived>
+  auto
+  act(const Eigen::MatrixBase<_EigenDerived> &v,
+      tl::optional<Eigen::Ref<Eigen::Matrix<Scalar, Dim, DoF>>> J_vout_m = {},
+      tl::optional<Eigen::Ref<Eigen::Matrix<Scalar, Dim, Dim>>> J_vout_v = {}) const
+  -> Eigen::Matrix<Scalar, Dim, 1>;
+
+  /**
+   * @brief Get the ajoint matrix of SO2 at this.
+   * @note See Eqs. (123).
+   */
+  Jacobian adj() const;
+
+  // SO2 specific functions
+
+  /**
+   * @brief Get the transformation matrix (2D isometry).
+   * @note T = | 0 t |
+   *           | 0 1 |
+   */
+  Transformation transform() const;
+
+
+protected:
+
+  using Base::coeffs_nonconst;
+};
+
+template <typename _Derived>
+typename RnBase<_Derived>::Transformation
+RnBase<_Derived>::transform() const
+{
+  Transformation T(Transformation::Identity());
+  T.template topRightCorner<Dim,1>() = coeffs();
+  return T;
+}
+
+template <typename _Derived>
+typename RnBase<_Derived>::LieGroup
+RnBase<_Derived>::inverse(OptJacobianRef J_minv_m) const
+{
+  if (J_minv_m)
+    J_minv_m->setIdentity()*=Scalar(-1);
+
+  return LieGroup(-coeffs());
+}
+
+template <typename _Derived>
+typename RnBase<_Derived>::Tangent
+RnBase<_Derived>::log(OptJacobianRef J_t_m) const
+{
+  if (J_t_m)
+    J_t_m->setIdentity();
+
+  return Tangent(coeffs());
+}
+
+template <typename _Derived>
+typename RnBase<_Derived>::Tangent
+RnBase<_Derived>::lift(OptJacobianRef J_t_m) const
+{
+  return log(J_t_m);
+}
+
+template <typename _Derived>
+template <typename _DerivedOther>
+typename RnBase<_Derived>::LieGroup
+RnBase<_Derived>::compose(
+    const LieGroupBase<_DerivedOther>& m,
+    OptJacobianRef J_mc_ma,
+    OptJacobianRef J_mc_mb) const
+{
+  using std::abs;
+
+  static_assert(
+    std::is_base_of<RnBase<_DerivedOther>, _DerivedOther>::value,
+    "Argument does not inherit from RnBase !");
+
+  static_assert(
+    RnBase<_DerivedOther>::Dim==_DerivedOther::Dim, "Dimension mismatch !");
+
+  if (J_mc_ma)
+    J_mc_ma->setIdentity();
+
+  if (J_mc_mb)
+    J_mc_mb->setIdentity();
+
+  return LieGroup(coeffs() + m.coeffs());
+}
+
+template <typename _Derived>
+template <typename _EigenDerived>
+// Eigen::Matrix<typename RnBase<_Derived>::Scalar, RnBase<_Derived>::Dim, 1>
+auto
+RnBase<_Derived>::act(const Eigen::MatrixBase<_EigenDerived> &v,
+                      tl::optional<Eigen::Ref<Eigen::Matrix<Scalar, Dim, DoF>>> J_vout_m,
+                      tl::optional<Eigen::Ref<Eigen::Matrix<Scalar, Dim, Dim>>> J_vout_v) const
+-> Eigen::Matrix<Scalar, Dim, 1>
+{
+  assert_vector_dim(v, Dim);
+
+  if (J_vout_m)
+  {
+    J_vout_m->setIdentity();
+  }
+
+  if (J_vout_v)
+  {
+    J_vout_v->setIdentity();
+  }
+
+  return v + coeffs();
```

Comment:
I know it's stupid and commutative, but I would have written here `coeffs() + v`;